x-pack/plugins/ml/public/application/supplied_configurations/supplied_configurations.tsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo, useState } from 'react';
+import type {
+  EuiSearchBarOnChangeArgs,
+  SearchFilterConfig,
+  FieldValueOptionType,
+} from '@elastic/eui';
+import {
+  EuiCallOut,
+  EuiCard,
+  EuiIcon,
+  EuiFlexGrid,
+  EuiFlexItem,
+  EuiSearchBar,
+  EuiSpacer,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import useMountedState from 'react-use/lib/useMountedState';
+import useMount from 'react-use/lib/useMount';
+import { isPopulatedObject } from '@kbn/ml-is-populated-object';
+import { useMlKibana } from '../contexts/kibana';
+import type { Module } from '../../../common/types/modules';
+import { LoadingIndicator } from '../components/loading_indicator';
+import { filterModules } from './utils';
+import { SuppliedConfigurationsFlyout } from './supplied_configurations_flyout';
+
+export function isLogoObject(arg: unknown): arg is { icon: string } {
+  return isPopulatedObject(arg) && Object.hasOwn(arg, 'icon');
+}
+
+export const SuppliedConfigurations = () => {
+  const {
+    services: {
+      mlServices: {
+        mlApi: { getDataRecognizerModule },
+      },
+    },
+  } = useMlKibana();
+
+  const [modules, setModules] = useState<Module[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [query, setQuery] = useState(EuiSearchBar.Query.MATCH_ALL);
+  const [error, setError] = useState<EuiSearchBarOnChangeArgs['error'] | null>(null);
+  const [isFlyoutVisible, setIsFlyoutVisible] = useState<boolean>(false);
+  const [selectedModuleId, setSelectedModuleId] = useState<string | undefined>();
+
+  const closeFlyout = () => setIsFlyoutVisible(false);
+  const isMounted = useMountedState();
+
+  /**
+   * Loads recognizer module configuration.
+   */
+  const loadModules = useCallback(async () => {
+    if (isMounted()) {
```

Comment:
the `isMounted` check needs to happen after the async call has returned. Making the check before means the component could still be unmounted by the time the modules have arrived.